sklearn/linear_model/tests/test_ridge.py

```diff
@@ -694,3 +695,16 @@ def test_n_iter():
         reg = Ridge(solver=solver, max_iter=1, tol=1e-1)
         reg.fit(X, y_n)
         assert_equal(reg.n_iter_, None)
+
+
+def test_ridge_fit_intercept_sparse():
+    X, y = make_regression(n_samples=1000, n_features=2, n_informative=2,
+                           bias=10., random_state=42)
+    X_csr = sp.csr_matrix(X)
+
+    dense = Ridge(alpha=1., tol=1.e-15, solver='sag', fit_intercept=True)
+    sparse = Ridge(alpha=1., tol=1.e-15, solver='sag', fit_intercept=True)
```

Comment:
Could you please also add the case where `solver != 'sag'` and check for the solver switch warning with `assert_warns`?